### ISSUE TITLE
Make site available only for authenticated users

### ISSUE THREAD

**Issue by dmitry-abramov** [MichaelSidnev/Nozzarella#2]
**Description**
As a user I should have acces to the site only after login. If user is not logged in it should be redirected to login form. When user logs in (becomes authenticated) he should be redirected to the list with cheeses.

Login page has two fields, email and password, and login button. Both fields are required. 

**Use cases**:
- not authenticate user opens any link on site. He is redirected to the login page
- not authenticated user enters valid email and password on login form and clicks login button. User becomes authenticated and redirected to the list of cheeses
- authentcated user opens login form. User is redirected to the list of cheeses
- not authenticated user clicks login button without fill up email and/or password field. Empty fields should be highlighted with red and error messages should appear with every empty field "Field is required.". It should be client side validation, no request should be sent to the server
- not authenticated user enter invalid pair of email and password. User should not be authenticated and stay on login form. Error message should appear "Email or password is invalid."

**Assumptions**
- there are one or several predefined users in **database**

**Out of scope**:
- registration of new users
- reset password
- change password
- logout
